runtime/storage/firebase-storage.js

```diff
@@ -114,11 +140,14 @@ export class FirebaseStorage {
 
     let currentSnapshot;
     await reference.once('value', snapshot => currentSnapshot = snapshot);
-    if (shouldExist != currentSnapshot.exists()) {
+    if (shouldExist == true && currentSnapshot.exists() == false) {
+      return null;
+    }
+    if (shouldExist == false && currentSnapshot.exists() == true) {
```

Comment:
Could use a single if for this?

 if (shouldExist !== 'unknown' && shouldExist != currentSnapshot.exists()) ...

Reply:
Done